Sources/FunctionalTableData/CellActions.swift

```diff
@@ -173,7 +277,8 @@ public struct CellActions {
 		canPerformAction: CanPerformAction? = nil,
 		canBeMoved: Bool = false,
 		visibilityAction: VisibilityAction? = nil,
-		previewingViewControllerAction: PreviewingViewControllerAction? = nil) {
+		previewingViewControllerAction: PreviewingViewControllerAction? = nil,
+		contextMenuConfiguration: ContextMenuConfiguration? = nil) {
```

Comment:
Why not allow passing in an instance of `UIContextMenuConfiguration` directly instead of having the `ContextMenuConfiguration` that then needs to be mapped to that? It seems like we are mostly duplicating the class...

Reply:
We are. The reason we don't allow for that is we can't store a `UIContextMenuConfiguration` in CellActions without making the entire struct only available to iOS 13+. I did look at storing it as an `AnyObject` but requiring all the extra casting felt wrong to me.

Reply:
I suggest we store it internally as an `Any?` and exposing it via a getter/setter with availability as it reduces the amount of code we have to manage and helps also keep us from having to maintain the mapping if/when Apple adds new properties to the `UIContextMenuConfiguration` type. Ideally we are able to stick as closely to vanilla UIKit as possible and only add the abstractions when absolutely necessary.

probably along the lines of:
```
private var _contextMenuConfiguration: Any?
@available(iOS 13.0, *)
public var contextMenuConfiguration: UIContextMenuConfiguration? {
    get { return _contextMenuConfiguration as! UIContextMenuConfiguration? }
    set { _contextMenuConfiguration = newValue }
}
```

We will also end up with an extra initializer. It would potentially need the ability to have `contextMenuConfiguration` passed in as an `Any?` in order to prevent having the call site worry about having two different init calls based on 13+ or <13. The init would be responsible for making sure that if the object is non-nil that it's type is correct.

Another option would be to leave it out of the `init` entirely and consumers would only be able to set via the property:
```
var actions = CellActions(...)
if #available(iOS 13, *) {
    actions.contextMenuConfiguration = UIContextMenuConfiguration(...)
}
```

Reply:
I did explore this solution, but there is an issue with it. Specifically, that we make use of the `identifier` field on the `UIContextMenuConfiguration`.

In the delegate function: `tableView(_: contextMenuConfigurationForRowAt: point:)` we invoke `asUIContextMenuConfiguration(with:)` on the Context Menu Configuration abstraction struct. This builds a `UIContextMenuConfiguration` and sets the identifier appropriately. We use the identifier in the following delegate function to identify what cell a context menu belongs to:
```Swift
@available(iOS 13.0, *)
public func tableView(_ tableView: UITableView, willPerformPreviewActionForMenuWith configuration: UIContextMenuConfiguration, animator: UIContextMenuInteractionCommitAnimating) {
	guard let indexPath = configuration.identifier as? IndexPath else { return }
	let cellConfig = data.sections[indexPath]
	animator.addCompletion {
		cellConfig?.actions.contextMenuConfiguration?.previewContentCommitter?(animator.previewViewController)
	}
}
```

`identifier` is a get only property on `UIContextMenuConfiguration`. If we store the `UIContextMenuConfiguration` directly in the CellActions struct, we will not be able to change the identifier. We instead have to rely on the consumer having properly set the identifier. This seems more error prone than handling it ourselves.

I'm not sure how else we could solve this problem, but by keeping the `ContextMenuConfiguration` abstraction struct. I would love to hear your thoughts on the matter.